### PR TITLE
feat: Laravel 9

### PR DESCRIPTION
**Pull request by atoff**
Alongside upgrading to Laravel 9, also adds support for VS Code Dev Containers

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM webdevops/php-nginx-dev:8.1
+
+ENV WEB_DOCUMENT_ROOT=/workspace/public
+
+RUN curl -sL https://deb.nodesource.com/setup_18.x | bash -
+RUN apt-get install -y nodejs
+
+RUN npm install -g yarn
```
Comment (AxonC) [inline .devcontainer/Dockerfile:8]: Line ending faff

File: .devcontainer/mysql_init/01-databases.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CREATE DATABASE IF NOT EXISTS `core`;
+CREATE DATABASE IF NOT EXISTS `cts`;
+
+GRANT ALL PRIVILEGES ON *.* TO 'root'@'%';
```
Comment (AxonC) [inline .devcontainer/mysql_init/01-databases.sql:4]: Line ending faff

**Review by AxonC — DISMISSED** [review]

**Comment by AxonC** [comment]
@atoff some conflicts following the permissions downgrade otherwise think this is good to merge

**Comment by atoff** [comment]
> @atoff some conflicts following the permissions downgrade otherwise think this is good to merge

So spatie permissions must be bumped to v5 to support Laravel 9. Will need to take a bit of time to bump locally, see if I can reproduce the issues we had in prod. If not, I reckon we bump permissions up to v5 and manually flush cache on deployment to see if it fixes the issue we saw.

**Comment by CalumTowers** [comment]
@atoff If you can resolve conflicts we can try that

**Review by CalumTowers — APPROVED** [review]

**Pull request merged**